benchmark/src/main/java/ai/timefold/solver/benchmark/config/PlannerBenchmarkConfig.java

```diff
@@ -543,168 +509,172 @@ public void setThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryCl
      *
      * @return null, a number or {@value #PARALLEL_BENCHMARK_COUNT_AUTO}.
      */
-    public String getParallelBenchmarkCount() {
+    public @Nullable String getParallelBenchmarkCount() {
         return parallelBenchmarkCount;
     }
 
-    public void setParallelBenchmarkCount(String parallelBenchmarkCount) {
+    public void setParallelBenchmarkCount(@Nullable String parallelBenchmarkCount) {
         this.parallelBenchmarkCount = parallelBenchmarkCount;
     }
 
-    public Long getWarmUpMillisecondsSpentLimit() {
+    public @NonNull Long getWarmUpMillisecondsSpentLimit() {
         return warmUpMillisecondsSpentLimit;
     }
 
-    public void setWarmUpMillisecondsSpentLimit(Long warmUpMillisecondsSpentLimit) {
+    public void setWarmUpMillisecondsSpentLimit(@NonNull Long warmUpMillisecondsSpentLimit) {
         this.warmUpMillisecondsSpentLimit = warmUpMillisecondsSpentLimit;
     }
 
-    public Long getWarmUpSecondsSpentLimit() {
+    public @NonNull Long getWarmUpSecondsSpentLimit() {
         return warmUpSecondsSpentLimit;
     }
 
-    public void setWarmUpSecondsSpentLimit(Long warmUpSecondsSpentLimit) {
+    public void setWarmUpSecondsSpentLimit(@NonNull Long warmUpSecondsSpentLimit) {
         this.warmUpSecondsSpentLimit = warmUpSecondsSpentLimit;
     }
 
-    public Long getWarmUpMinutesSpentLimit() {
+    public @NonNull Long getWarmUpMinutesSpentLimit() {
         return warmUpMinutesSpentLimit;
     }
 
-    public void setWarmUpMinutesSpentLimit(Long warmUpMinutesSpentLimit) {
+    public void setWarmUpMinutesSpentLimit(@NonNull Long warmUpMinutesSpentLimit) {
         this.warmUpMinutesSpentLimit = warmUpMinutesSpentLimit;
     }
 
-    public Long getWarmUpHoursSpentLimit() {
+    public @NonNull Long getWarmUpHoursSpentLimit() {
         return warmUpHoursSpentLimit;
     }
 
-    public void setWarmUpHoursSpentLimit(Long warmUpHoursSpentLimit) {
+    public void setWarmUpHoursSpentLimit(@NonNull Long warmUpHoursSpentLimit) {
         this.warmUpHoursSpentLimit = warmUpHoursSpentLimit;
     }
 
-    public Long getWarmUpDaysSpentLimit() {
+    public @NonNull Long getWarmUpDaysSpentLimit() {
```

Comment:
I'm afraid this pattern is not correct. :-(

- Getters in configs have to be `Nullable`, because the fields may hold `null`. `null` here means that the user chose not to specify. (Why else would we choose `Long` instead of `long`, avoiding reference types altogether?)
- Setters likewise have to be able to accept `null`, because the solver code needs to be able to express that something should explicitly be set to its default value.
- It is only the withers which should not accept null and never return it, because we understand withers to mean that the user is making an explicit choice, and therefore `null` no longer applies. 

This is an unfortunate miscommunication. We have discussed this topic in comments above - even after re-reading my answer, I still think I said the same thing as I am saying here.

Sadly, I can not accept this and the annotations in configs need to be reviewed with the above in mind. :-( Interestingly, it seems like not all configs use this wrong pattern - some configs properly assign nullability as we agreed.

Maybe this file was annotated before our agreement?

Reply:
This is a mistake on my part. Seems like some error using multi-cursor and then pasting the wrong value. I'll carefully review getters and setters myself to make sure this didn't happen more often. Thanks for catching that. 

In general, I'm pretty sure I adhered to the pattern as discussed (getters/setters nullable, withers taking nonnull params)